sympy/discrete/convolution.py

```diff
@@ -0,0 +1,192 @@
+"""
+Convolution (using FFT, NTT),
+Arithmetic Convolution (bitwise XOR, AND, OR), Subset Convolution
+"""
+from __future__ import print_function, division
+
+from sympy.core import S
+from sympy.core.compatibility import range, as_int
+from sympy.core.function import expand_mul
+from sympy.discrete.transforms import fft, ifft, ntt, intt
+
+
+def convolution(a, b, **hints):
+    """
+    Performs convolution by determining the type of desired
+    convolution using hints.
+
+    If no hints are given, linear convolution is performed using
+    FFT.
+
+    Parameters
+    ==========
+
+    a, b : iterables
+        The sequences for which convolution is performed.
+    hints : dict
```

Comment:
`hints` isn't really a dict from the user point of view. Rather, there are keyword arguments. So it would be better to just list the keyword arguments on the function signature directly (`**kwargs` should only be used when arbitrary keyword arguments are required).